### DISCUSSION TITLE
Add support for Telegram group chat topics

### DISCUSSION THREAD

**Discussion by 0xhirugohan** [d99kris/nchat#321] [Ideas]
Topics in Telegram Group is announced around Nov 2022 https://telegram.org/blog/topics-in-groups-collectible-usernames

Currently when accessing Group with Topics in nchat, it only displays chat as its not enabled a Topic. But when you reply to one of the chat in a certain topic, it will send your message to that specific topic as well, so it works.

The problem is, I would like to know which chat are from which topic.

Currently:

```
User A (10:00) :white_check_mark: 
A certain message in Topic A

User B (10:00) :white_check_mark: 
A certain message in Topic A

User C (10:00) :white_check_mark: 
A certain message in Topic C
```

And I am expecting something like this:

```
User A (10:00) in Topic A :white_check_mark: 
A certain message in Topic A

User B (10:00) in Topic A :white_check_mark: 
A certain message in Topic A

User C (10:00) in Topic C :white_check_mark: 
A certain message in Topic C
```
Is that possible in the current version of nchat? If so, how to do that? Thank you

**Comment by d99kris** [comment]
Topics is currently not supported by nchat. I will convert this discussion into a feature request and see what can be done to support it. At a first glance I think the main issue to resolve is to provide an intuitive UI for reading and replying to the topics (even without replying to a specific message in a topic), so I need to think a bit about that and perhaps compare with other Telegram clients.

**Reply by d99kris** [reply]
Some UI suggestion for topics was posted in #333 but I have not yet reviewed that.
